sheep/object_list_cache.c

```diff
@@ -111,9 +112,17 @@ int get_obj_list(const struct sd_req *hdr, struct sd_rsp *rsp, void *data)
 	if (obj_list_cache.tree_version == obj_list_cache.buf_version)
 		goto out;
 
+	/* Update obj_list_cache.buf indirectly to keep previous pointer */
+	newbuf = realloc(obj_list_cache.buf,
+			 obj_list_cache.cache_size * sizeof(uint64_t));
+	if (!newbuf && errno == ENOMEM) {
+		sd_rw_unlock(&obj_list_cache.lock);
```

Comment:
This error handling is a little bit complicated because the function `get_obj_lsit()` has both of goto style error handling and direct error handling in a failed case. Could you unify them?